STAR.Web.Tests/ContractorControllerTests.cs

```diff
@@ -30,5 +32,51 @@ public class ContractorControllerTests {
 
             Assert.That(model, Has.Count.EqualTo(0));
         }
+
+        [Test]
+        public void PostBlankFirstNameYieldsInvalid() {
+            var mockContext = new Mock<StarContext>();
+            var controller = new ContractorController(mockContext.Object);
+            var contractor = new PostContractorViewModel();
+            contractor.Id = 1;
+            contractor.LastName = "Judd";
+            contractor.SkillIds = "1,2,3";
+
+            controller.ModelState.Clear();
+            var validationContext = new ValidationContext(contractor, null, null);
+            var validationResults = new List<ValidationResult>();
+            Validator.TryValidateObject(contractor, validationContext, validationResults, true);
+            foreach (var v in validationResults) {
+                foreach (var name in v.MemberNames) {
+                    controller.ModelState.AddModelError(name, v.ErrorMessage);
+                }
+            }
+
+            var result = controller.Details(contractor) as ViewResult;
+            Assert.False(result.ViewData.ModelState.IsValid);
+        }
+
+        [Test]
+        public void PostBlankLastNameYieldsInvalid() {
+            var mockContext = new Mock<StarContext>();
+            var controller = new ContractorController(mockContext.Object);
+            var contractor = new PostContractorViewModel();
+            contractor.Id = 1;
+            contractor.FirstName = "Jim";
+            contractor.SkillIds = "1,2,3";
+
+            controller.ModelState.Clear();
```

Comment:
Extract a function for setting up the validation of the model in the unit tests, with an eye on reusing this for other controllers as well.